pkg/clusteroperator/clusteroperator.go

```diff
@@ -0,0 +1,210 @@
+
+	return &builder, nil
+}
+
+// Exists checks whether the given ClusterOperator exists.
+func (builder *ClusterOperatorBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if ClusterOperator %s exists", builder.Definition.Name)
+
+	var err error
+	builder.Object, err = builder.Get()
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// Get fetches the defined ClusterOperator from the cluster.
+func (builder *ClusterOperatorBuilder) Get() (*v1.ClusterOperator, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Getting ClusterOperator %s", builder.Definition.Name)
+
+	module := &v1.ClusterOperator{}
+
+	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
+		Name: builder.Definition.Name,
+	}, module)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return module, err
+}
+
+// IsAvailable check if the ClusterOperator is Available.
+func (builder *ClusterOperatorBuilder) IsAvailable() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Verify %s clusteroperators availability", builder.Definition.Name)
+
+	var err error
+	builder.Object, err = builder.apiClient.ClusterOperators().Get(
+		context.Background(), builder.Definition.Name, metaV1.GetOptions{})
+
+	if err != nil {
+		return false
+	}
+
+	for _, condition := range builder.Object.Status.Conditions {
+		if condition.Type == "Available" {
+			return condition.Status == isTrue
+		}
+	}
+
+	return false
+}
+
+// IsDegraded check if the ClusterOperator is Degraded.
+func (builder *ClusterOperatorBuilder) IsDegraded() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Verify %s clusteroperators is degraded", builder.Definition.Name)
+
+	var err error
+	builder.Object, err = builder.apiClient.ClusterOperators().Get(
+		context.Background(), builder.Definition.Name, metaV1.GetOptions{})
+
+	if err != nil {
+		return false
+	}
+
+	for _, condition := range builder.Object.Status.Conditions {
+		if condition.Type == "Degraded" {
+			return condition.Status == isTrue
+		}
+	}
+
+	return false
+}
+
+// WaitUntilAvailable waits for timeout duration or until ClusterOperator is Available.
+func (builder *ClusterOperatorBuilder) WaitUntilAvailable(timeout time.Duration) error {
+	return builder.WaitUntilInStatus("Available", timeout)
+}
+
+// WaitUntilProgressing waits for timeout duration or until ClusterOperator is Progressing.
+func (builder *ClusterOperatorBuilder) WaitUntilProgressing(timeout time.Duration) error {
+	return builder.WaitUntilInStatus("Progressing", timeout)
+}
+
+// WaitUntilInStatus waits for timeout duration or until ClusterOperator gets to a specific status.
+func (builder *ClusterOperatorBuilder) WaitUntilInStatus(
+	status v1.ClusterStatusConditionType, timeout time.Duration) error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
```

Comment:
Please check that cluster operator exists first

Reply:
I use validate() method instead

Reply:
done